modules/installation-about-custom.adoc

```diff
@@ -1,95 +1,51 @@
-than the interactive prompts and comes pre-populated with default values.
-`manifest-templates`:: These are the unrendered Kubernetes manifest templates
-that `manifests` target uses. This target is unstable.
-`manifests`::
-You can generate the Kubernetes manifests that will be installed on the cluster.
-This target is unstable.
-`ignition-configs`::
-These files contain the Ignition configurations for the bootstrap, master, and
-worker machines.
-`cluster`::
-This target provisions the cluster and its associated infrastructure.
-
-The following targets can be destroyed by the installation program:
-
-`cluster`::
-This destroys the created cluster and its associated infrastructure.
-`bootstrap`::
-This destroys the bootstrap infrastructure.
-
-[discrete]
-=== Multiple invocations
-
-Because you must run the installation program to generate the assets that you need to
-customize your installation, you can run the installation program multiple times. The
-installation state is stored in a hidden file in the asset directory which 
-contains all of the intermediate artifacts. Storing the artifacts in this way
-lets you pause installation and modify the intermediate artifacts.
-
-For example, if you wanted to create a different number of worker machines, you
-must first run the installation program with the `install-config` target with the
-`openshift-install create install-config` command. After you provide the required
-parameters to the interactive installer, it writes the installation
-configuration into the target directory. You can then modify the installation
-configuration and run the installation program with the `cluster` target with the
-`openshift-install create cluster` command. The installation program reads the install
-configuration from disk, removes it from the target directory, and creates a
-cluster from the provided configuration.
+You can use the {product-title} installation program to customize four levels
+of the program:
+
+* {product-title} itself
+* The cluster platform
+* Kubernetes
+* The cluster operating system
+
+Changes to {product-title} and its platform are managed and supported, but
+changes to Kubernetes and the cluster operating system are not. If you customize
+unsupported levels program levels, future installation and upgrades might fail.
+
+When you select values for the prompts that the installation program presents,
+you customize {product-title}. You can further modify the cluster platform
+by modifying the `install-config.yaml` file that the installation program
+uses to deploy your cluster. In this file, you can make changes like setting the
+number of machines that the control plane uses, the type of virtual machine
+that the cluster deploys, or the CIDR range for the Kubernetes service network.
+
+It is possible, but not supported, to modify the Kubernetes objects that are injected into the cluster.
+A common modification is additional manifests in the initial installation. 
```

Comment:
> Is there a good way for users to identify which Kubernetes objects are safe to change?

You can find out what the CVO manages by inspecting the update payload, but that doesn't cover indirect dependencies (e.g. objects pushed by CVO-managed operators).  And it looks like Kubernetes considered and then [rejected][1] recording creator information.  As long as they stick to namespaces/projects they create, they'll be fine.  For namespaces we create and cluster-scoped objects, I'm not sure what an easy way would be to check whether a change would be supported short of making the edit and seeing if it gets stomped ;).  Maybe @abhinavdahiya or @smarterclayton have some more elegant ideas.

[1]: https://github.com/kubernetes/kubernetes/issues/13306#issuecomment-351297076

Reply:
> And it looks like Kubernetes considered and then rejected recording creator information.

Although creators can [set `ownerReference` explicitly if they need to][1].  Maybe we just need to push more operators to do that (if they aren't already).

[1]: https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#owners-and-dependents

Reply:
@jianlinliu, are you testing any modifications to Kubernetes manifests during installation?

Reply:
Yeah, we had some testing to modify Kubernetes manifests for customized install until some discussion in upstream repo - https://github.com/openshift/installer/pull/1088/files#r250505048. Since that, we lower the testing of any modifications to Kubernetes manifests, and any issue found during testing is not treated as a blocker issue any more. Because QE also thought modification via manifest and ignition is not officially supported. 
I agree with you, if we support it, we have to tell user which Kubernetes objects are safe to change, or a better approach is to prompt the modification to install-config level, we treat everything in install-config is officially supported to allow modification. 
@qinpingli pls follow up this discussion, make sure what we tested is on the same page like what we published on our official doc.

Reply:
Thank you @jianlinliu! @qinpingli, please check during the E2E tests and follow up. Per @vikram-redhat, we're merging frequently instead of holding docs for 2/22.  
@sferich888, are we planning to support modifying Kubernetes manifests during installation at 4.0 GA?